libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2078,6 +2086,75 @@ void TypeChecker::typeCheckABIEncodeFunctions(
 	}
 }
 
+void TypeChecker::typeCheckABIEncodeCallFunction(FunctionCall const& _functionCall)
+{
+	vector<ASTPointer<Expression const>> const& arguments = _functionCall.arguments();
+
+	// Expecting first argument to be the function pointer and second to be a tuple.
+	if (arguments.size() != 2)
+	{
+		m_errorReporter.typeError(
+			6219_error,
+			_functionCall.location(),
+			"Expected two arguments: a function pointer followed by a tuple."
+		);
+		return;
+	}
+
+	auto const functionPointerType = dynamic_cast<FunctionTypePointer>(type(*arguments.front()));
+	auto const argumentTuple = dynamic_cast<TupleType const*>(type(*arguments[1]));
+
+	if (!functionPointerType)
+	{
+		m_errorReporter.typeError(
+			5511_error,
+			_functionCall.location(),
+			"Expected first argument to be a function pointer, not \"" +
+			type(*arguments.front())->canonicalName() +
+			"\"."
+		);
+		return;
+	}
+
+	// There are no checks we can do if all types are accepted
+	if (functionPointerType->takesArbitraryParameters())
```

Comment:
I think we should ensure / assert that it does not take arbitrary parameters.
Can you add a type check that the function kind is `FunctionType::Kind::External`?